optimizely/event_builder.py

```diff
@@ -35,78 +35,84 @@ class BaseEventBuilder(object):
 
   def __init__(self, config):
     self.config = config
-    self.params = {}
 
   @abstractproperty
   class EventParams(object):
     pass
 
-  def _add_project_id(self):
-    """ Add project ID to the event. """
+  def _get_project_id(self):
+    """ Get project ID. """
 
-    self.params[self.EventParams.PROJECT_ID] = self.config.get_project_id()
+    return self.config.get_project_id()
 
-  def _add_account_id(self):
-    """ Add account ID to the event. """
+  def _get_account_id(self):
+    """ Get account ID. """
 
-    self.params[self.EventParams.ACCOUNT_ID] = self.config.get_account_id()
+    return self.config.get_account_id()
 
   @abstractmethod
-  def _add_user_id(self, user_id):
-    """ Add user ID to the event.
+  def _get_user_id(self, user_id):
+    """ Get user ID.
 
     Args:
       user_id: ID of the user.
     """
     pass
 
   @abstractmethod
-  def _add_attributes(self, attributes):
-    """ Add attribute(s) information to the event.
+  def _get_attributes(self, attributes):
+    """ Get attribute(s) information.
 
     Args:
       attributes: Dict representing user attributes and values which need to be recorded.
     """
     pass
 
   @abstractmethod
-  def _add_source(self):
-    """ Add source information to the event. """
+  def _get_source(self):
+    """ Get source information. """
     pass
 
-  def _add_revision(self):
-    """ Add datafile revision information to the event. """
+  def _get_revision(self):
+    """ Get datafile revision information. """
     pass
 
-  def _add_anonymize_ip(self):
-    """ Add IP anonymization bool to the event """
+  def _get_anonymize_ip(self):
+    """ Get IP anonymization bool """
 
-    self.params[self.EventParams.ANONYMIZE_IP] = self.config.get_anonymize_ip_value()
+    return self.config.get_anonymize_ip_value()
 
   @abstractmethod
   def _get_time(self):
-    """ Get time in milliseconds to be added to the event.
+    """ Get time in milliseconds to be added.
 
     Returns:
       Current time in milliseconds.
     """
 
     return int(round(time.time() * 1000))
 
-  def _add_common_params(self, user_id, attributes):
-    """ Add params which are used same in both conversion and impression events.
+  def _get_common_params(self, user_id, attributes):
```

Comment:
Please restore to the previous state.

Reply:
this change was needed because existing code is not thread safe & all other SDKs using the same approach.

Reply:
Can you elaborate on the thread safety aspect?

Reply:
in the old code, when create_conversion_event is called, it reset params property
https://github.com/optimizely/python-sdk/blob/master/optimizely/event_builder.py#L293

and if createconversion event is called in parallel twice using same optimizely object, the later one will overwrite first params values because param is global in the EventBuilder class. and EventBuilder is initialized only once in the Optimizely class.
https://github.com/optimizely/python-sdk/blob/master/optimizely/optimizely.py#L82

Reply:
Ok. Then keep the change and update the docstrings to reflect what is being returned in each method which you changed.